pycparser/c_generator.py

```diff
@@ -289,23 +278,55 @@ def visit_NamedInitializer(self, n):
     def visit_FuncDecl(self, n):
         return self._generate_type(n)
 
-    def _generate_struct_union(self, n, name):
-        """ Generates code for structs and unions. name should be either
-            'struct' or union.
+    def _generate_struct_union_enum(self, n, name):
+        """ Generates code for structs, unions, and enums. name should be
+            'struct', 'union', or 'enum'.
         """
+        if name in ('struct', 'union'):
+            members = n.decls
+            body_function = self._generate_struct_union_body
+        elif name in ('enum',):
+            if n.values is None:
+                members = ()
+            else:
+                members = n.values.enumerators
+            body_function = self._generate_enum_body
+
         s = name + ' ' + (n.name or '')
-        if n.decls:
+        if members:
             s += '\n'
             s += self._make_indent()
             self.indent_level += 2
             s += '{\n'
-            for decl in n.decls:
-                s += self._generate_stmt(decl)
+            s += body_function(members)
             self.indent_level -= 2
             s += self._make_indent() + '}'
         return s
 
-    def _generate_stmt(self, n, add_indent=False):
+    def _generate_struct_union_body(self, members):
+        return ''.join(self._generate_stmt(decl) for decl in members)
+
+    def _generate_enum_body(self, members):
+        s = []
+
+        punctuation = ','
+
+        for i, value in enumerate(members):
+            if i == len(members) - 1:
+                punctuation = ''
+
+            s.append(self._generate_stmt(
```

Comment:
I wonder why you need to build an AST node again here, why won't just emitting the text do?

Reply:
I was trying to make it look like how `struct` and `union` were done, but failing.  I removed the punctuation parameter from `_generate_stmt()` and added `visit_Enumerator()`.  Perhaps this will fit better with the existing code.